geary-papermc-spawning/src/main/kotlin/com/mineinabyss/geary/papermc/spawning/choosing/SpreadChunkChooser.kt

```diff
@@ -0,0 +1,87 @@
+package com.mineinabyss.geary.papermc.spawning.choosing
+
+import com.mineinabyss.geary.papermc.spawning.config.SpreadSpawnConfig
+import com.mineinabyss.geary.papermc.spawning.database.dao.SpawnLocationsDAO
+import com.mineinabyss.geary.papermc.spawning.spread_spawn.SpreadSpawner
+import me.dvyy.sqlite.Database
+import org.bukkit.Chunk
+import org.bukkit.Location
+import org.bukkit.World
+import org.bukkit.util.BoundingBox
+import org.bukkit.util.Vector
+import kotlin.random.Random
+
+class SpreadChunkChooser {
+
+    suspend fun chooseChunkInBB(bb: BoundingBox, spawner: SpreadSpawner, config: SpreadSpawnConfig): Location? {
```

Comment:
This function is pretty beefy, you don't need to split it up per se but a javadoc comment above it explaining on a high level how the system works could be really helpful. It would be nice to see these on some other important/large functions too, but this one is particularly important.